backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandlerCacheManager.java

```diff
@@ -0,0 +1,44 @@
+package co.airy.core.api.components.installer;
+
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1Job;
+
+import org.slf4j.Logger;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import co.airy.log.AiryLoggerFactory;
+
+@Component
+public class InstallerHandlerCacheManager {
+
+    private static final Logger log = AiryLoggerFactory.getLogger(InstallerHandlerCacheManager.class);
+
+    private final HelmJobHandler helmJobHandler;
+    private final ApiClient apiClient;
+    private final InstalledComponentsHandler installedComponentsHandler;
+
+    InstallerHandlerCacheManager(
+            ApiClient apiClient,
+            HelmJobHandler helmJobHandler,
+            InstalledComponentsHandler installedComponentsHandler) {
+        this.apiClient = apiClient;
+        this.helmJobHandler = helmJobHandler;
+        this.installedComponentsHandler = installedComponentsHandler;
+    }
+
+    @Async("threadPoolTaskExecutor")
+    public void resetCacheAfterJob(String jobName) {
+        try {
+            final V1Job job = helmJobHandler.getJobByName(jobName);
+            final CoreV1Api api = new CoreV1Api(apiClient);
+            helmJobHandler.waitForCompletedStatus(api, job);
```

Comment:
As install/uninstall can take some time, it is good that we wait for the job to be completed successfully before resetting the cache :+1: